upload/upload.go

```diff
@@ -129,26 +129,23 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 			return err
 		}
 
-		// Check if files exists in S3
-		var listPrefix string
-		if targetDir != "" {
-			listPrefix = targetDir + "/" + outFiles[k]
+		if *forceOverwrite {
+			fmt.Println("force-overwrite flag provided, continuing...")
 		} else {
-			listPrefix = outFiles[k]
-		}
-		fileExists, err := helpers.ListFiles(*config, listPrefix)
-		if err != nil {
-			log.Error("Couldn't get the file list ", err)
-		}
-		if len(fileExists.Contents) > 0 {
-			if aws.StringValue(fileExists.Contents[0].Key) == filepath.Clean(config.AccessKey+"/"+targetDir+"/"+outFiles[k]) {
+			// Check if files exists in S3
+			listPrefix := outFiles[k]
+			if targetDir != "" {
+				listPrefix = targetDir + "/" + outFiles[k]
+			}
+
+			fileExists, err := helpers.ListFiles(*config, listPrefix)
+			if err != nil {
+				return fmt.Errorf("listing uploaded files: %s", err.Error())
+			}
+			if len(fileExists.Contents) > 0 && aws.StringValue(fileExists.Contents[0].Key) == filepath.Clean(config.AccessKey+"/"+targetDir+"/"+outFiles[k]) {
 				fmt.Printf("File %s is already uploaded!\n", filepath.Base(filename))
-				if !*forceOverwrite {
-					fmt.Println("Quitting...")
 
-					return errors.New("file already uploaded")
-				}
-				fmt.Println("force-overwrite flag provided, continuing...")
+				return errors.New("file already uploaded")
```

Comment:
How are these changes related to the token expiring?

Reply:
Admittedly, they're not.

Reply:
Then it could be included in a scoped refactoring PR (targeting only the upload functionality).
Since it is not remotely related to this PR it should not be here.